### PR TITLE
feat: limit orders input validation and quote error rendering

### LINKED ISSUE #8109: Add input validation and error handling for limit orders
### Overview

- [x] Input validation (balance, connected accounts etc)
- [x] Remove native assets from sell/buy asset menus
- [x] Reset input amount on asset change
- [x] Prevent selecting the same buy and sell asset
- [x] Reset receive address when assets change
- [x] Render errors from the quote on the confirmation button in red, similar to trade input page
- [x] Prevent proceeding with invalid orders by disabling the confirmation button

### References and additional details

.

### Acceptance Criteria

See above.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Performs validation of user input and renders quote errors, if there are any. Prevents users from proceeding with invalid limit orders.

## Issue (if applicable)

closes #8109

## Risk

> High Risk PRs Require 2 approvals

Low risk, this makes no changes to anything outside of limit orders, which is behind a feature flag and still work-in-progress.

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

### Operations

- [x] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

## Screenshots (if applicable)

Quick jam showing preview button disabled before quote, and balance validation:
https://jam.dev/c/b835bceb-9de3-4287-bfa7-57dc7f562007

### CODE REVIEW

File: src/assets/translations/en/main.json [hunk 1]
```diff
@@ -1046,6 +1047,12 @@
       "sevenDays": "7 days",
       "twentyEightDays": "28 days",
       "custom": "Custom"
+    },
+    "errors": {
+      "nativeSellAssetNotSupported": "Native sell asset not supported",
```
Comment (gomesalexandre) [inline src/assets/translations/en/main.json:1052]: Tagging @shapeshift/product here for copy follow-up, IMO, we probably don't want to tell users about "native assets" but be more explicit e.g something like "Cannot use ETH as a limit order sell asset" or similar

File: src/components/MultiHopTrade/components/LimitOrder/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import type { SerializedError } from '@reduxjs/toolkit'
+import type { CowSwapError } from '@shapeshiftoss/swapper'
+import { OrderError } from '@shapeshiftoss/types/dist/cowSwap'
+import type { InterpolationOptions } from 'node-polyglot'
+import { assertUnreachable } from 'lib/utils'
+
+export const isCowSwapError = (maybeCowSwapError: any): maybeCowSwapError is CowSwapError => {
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/LimitOrder/helpers.ts:7]:
```suggestion
export const isCowSwapError = (maybeCowSwapError: CowSwapError | SerializedError | undefined): maybeCowSwapError is CowSwapError => {
```

**Review by gomesalexandre — APPROVED** [review]
Tested locally, LGTM!

https://jam.dev/c/d1157626-b900-4dbf-ad9a-faef764130ac

Noting approval error-handling is implemented but not working just yet, which makes sense as actual allowance checks aren't present, which is expected!

One comment re: types to tackle at your convenience, feel free to merge 🍰

File: src/components/MultiHopTrade/components/LimitOrder/helpers.ts (hunk 2 above)
Comment (woodenfurniture) [reply src/components/MultiHopTrade/components/LimitOrder/helpers.ts:7]: 🧠 will do this ser

**Pull request merged**